langlib/lang.value/src/main/ballerina/value.bal

```diff
@@ -177,6 +214,11 @@ public isolated function fromBalString(string s) returns anydata|error = @java:M
 # immutable values.
 # This panics if parameter `v` has cycles.
 #
+# ```ballerina
+# record {|int index; string name;|} student = {index: 12, name: "John"};
```

Comment:
Shall we define a separate type instead of using an anonymous record? Just to keep the sample simple.

Reply:
if we define a separate type, we have to wrap this in a function ryt?
it also reduce the simplicity